packages/aws-cdk/lib/api/hotswap/lambda-functions.ts

```diff
@@ -64,43 +69,94 @@ async function isLambdaFunctionCodeOnlyChange(
+  let foundDifference = false;
+  let tagUpdates: { [tag: string]: string | TagDeletion } = {};
   // Make sure only the code in the Lambda function changed
   const propertyUpdates = change.propertyUpdates;
   for (const updatedPropName in propertyUpdates) {
     const updatedProp = propertyUpdates[updatedPropName];
-    for (const newPropName in updatedProp.newValue) {
-      switch (newPropName) {
-        case 'S3Bucket':
-          foundCodeDifference = true;
-          s3Bucket = await evaluateCfnTemplate.evaluateCfnExpression(updatedProp.newValue[newPropName]);
-          break;
-        case 'S3Key':
-          foundCodeDifference = true;
-          s3Key = await evaluateCfnTemplate.evaluateCfnExpression(updatedProp.newValue[newPropName]);
-          break;
-        default:
-          return ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT;
-      }
+
+    switch (updatedPropName) {
+      case 'Code':
+        for (const newPropName in updatedProp.newValue) {
+          switch (newPropName) {
+            case 'S3Bucket':
+              foundDifference = true;
+              s3Bucket = await evaluateCfnTemplate.evaluateCfnExpression(updatedProp.newValue[newPropName]);
+              break;
+            case 'S3Key':
+              foundDifference = true;
+              s3Key = await evaluateCfnTemplate.evaluateCfnExpression(updatedProp.newValue[newPropName]);
+              break;
+            default:
+              return ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT;
+          }
+        }
+        break;
+      case 'Tags':
+        /*
+         * Tag updates are a bit odd; they manifest as two lists, are flagged only as
+         * `isDifferent`, and we have to reconcile them.
+         */
+
+        if (updatedProp !== undefined) {
+          foundDifference = foundDifference || updatedProp.isDifferent;
+
+          updatedProp.newValue.forEach((tag: CfnDiffTagValue) => {
+            tagUpdates[tag.Key] = tag.Value;
+          });
+
+          updatedProp.oldValue.forEach((tag: CfnDiffTagValue) => {
+            if (tagUpdates[tag.Key] === undefined) {
+              tagUpdates[tag.Key] = TagDeletion.DELETE;
+            }
+          });
+        }
+        break;
+      default:
+        return ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT;
     }
   }
 
-  return foundCodeDifference
+  return foundDifference
     ? {
-      s3Bucket,
-      s3Key,
+      code: {
+        s3Bucket,
+        s3Key,
+      },
+      tags: {
+        tagUpdates,
+      },
     }
     : ChangeHotswapImpact.IRRELEVANT;
 }
 
+interface CfnDiffTagValue {
+  readonly Key: string;
+  readonly Value: string;
+}
+
 interface LambdaFunctionCode {
   readonly s3Bucket: string;
   readonly s3Key: string;
 }
 
+enum TagDeletion {
+  DELETE = -1,
+}
+
+interface LambdaFunctionTags {
+  readonly tagUpdates: { [tag : string] : string | TagDeletion };
+}
+
+interface LambdaFunctionChange {
+  code?: LambdaFunctionCode,
+  tags?: LambdaFunctionTags,
```

Comment:
Are these actually optional? Feels like they should be required.

Oh, and also `readonly` 🙂.

Reply:
That actually brought my attention to an inconsistency above. Expect the next rev to tidy those up a bit.